packages/core/src/client.ts

```diff
@@ -648,18 +536,21 @@ export class SkipRouter {
     );
 
     if (!accountFromSigner) {
-      throw new Error("Failed to retrieve account from signer");
+      throw new Error(
+        "signMultiChainMessageDirect error: failed to retrieve account from signer",
+      );
     }
 
     const message = getEncodeObjectFromMultiChainMessage(multiChainMessage);
 
-    const pubkey = encodePubkey(
-      encodeSecp256k1Pubkey(accountFromSigner.pubkey),
-    );
+    const algo = `${accountFromSigner.algo}`;
+    const isEthSecp256k1 = algo === "eth_secp256k1" || algo === "ethsecp256k1";
 
-    if ((accountFromSigner.algo as string) === "ethsecp256k1") {
-      pubkey.typeUrl = "/ethermint.crypto.v1.ethsecp256k1.PubKey";
-    }
+    const pubKey = isEthSecp256k1
+      ? encodeEthSecp256k1Pubkey(accountFromSigner.pubkey)
+      : encodeSecp256k1Pubkey(accountFromSigner.pubkey);
+
+    const encodedPubKey = encodePubkey(pubKey);
```

Comment:
As the reader, its strange to see we're calling an "encode[SpecificKindof]PubKey" function then invoking encodePubkey on the output of that first function. 

(Not saying its wrong, but it just looks awkward)

Reply:
- first pass is to verify algo by checking first byte and length, then returns a `PubKey` type with `{ type: string; value: base64 encoded string }`
- then `PubKey` is encoded to `Any` type with its respective `typeUrl` and `value`

Awkward, indeed.